File: Android_app/pioneer/src/main/java/com/ABC/pioneer/sensor/payload/Crypto/PioneerHash.java
package com.ABC.pioneer.sensor.payload.Crypto;

public class PioneerHash {
    Digest digest;

    public PioneerHash()
    {
        digest = new SM3Digest();
    }

    public void generateHash(byte[] message,byte[] hash)
    {
        generateHash(message,0,message.length,hash,0);
    }

    public void generateHash(byte[] message,int inoff,int len,byte[] hash,int outoff)
    {
        digest.update(message,inoff,len);
        digest.doFinal(hash,outoff);
    }
}

File: IOS_app/Pioneer/Pioneer Framework/Pioneer/Pioneer/Crypto/PioneerHash.swift
//
//  PioneerHash.swift
//  Pioneer
//
//  Created by Beh on 2021/6/3.
//

import Foundation

public class PioneerHash
{
    var digest:Digest
    
    init()
    {
        digest = SM3Digest()
    }
    
    public func generateHash(from message:[UInt8],to hash:inout [UInt8])
    {
        generateHash(from: message, inOff: 0, len: message.count, hash: &hash, outOff: 0)
    }
    
    public func generateHash(from message:[UInt8],inOff:Int,len:Int,hash:inout [UInt8],outOff:Int)
    {
        digest.update(inbytes: message, inOff: inOff, inLen: len)
        _ = digest.doFinal(outbytes: &hash, outOff: outOff)
    }
}
